src/ServiceControl/MessageFailures/Api/EditFailedMessages.cs

```diff
@@ -0,0 +1,151 @@
+                 * one with the same MessageID
+                 */
+
+                if (LockedHeaderModificationValidator.Check(GetEditConfiguration().LockedHeaders, edit.MessageHeaders.ToDictionary(x => x.Key, x => x.Value), failedMessage.ProcessingAttempts.Last().Headers))
+                {
+                    logging.WarnFormat("Locked headers have been modified on the edit-retry for MessageID {0}.", failedMessageId);
+                    return HttpStatusCode.BadRequest;
+                }
+
+                if (string.IsNullOrWhiteSpace(edit.MessageBody) || edit.MessageHeaders == null)
+                {
+                    logging.WarnFormat("There is no message body on the edit-retry for MessageID {0}.", failedMessageId);
+                    return HttpStatusCode.BadRequest;
+                }
+
+                // Encode the body in base64 so that the new body doesn't have to be escaped
+                var base64String = Convert.ToBase64String(Encoding.UTF8.GetBytes(edit.MessageBody));
+                await Bus.SendLocal(new EditAndSend
+                {
+                    FailedMessageId = failedMessageId,
+                    NewBody = base64String,
+                    NewHeaders = edit.MessageHeaders.ToDictionary(kvp => kvp.Key, kvp => kvp.Value)
+                }).ConfigureAwait(false);
+
+                return HttpStatusCode.Accepted;
+            };
+        }
+
+        public LockedHeaderModificationValidator LockedHeaderModificationValidator { get; set; }
+
+        EditConfigurationModel GetEditConfiguration()
+        {
+            return new EditConfigurationModel
+            {
+                Enabled = Settings.AllowMessageEditing,
+                LockedHeaders = new[]
+                {
+                    Headers.MessageId,
+                    Headers.SagaId,
+                    Headers.CorrelationId,
+                    Headers.ControlMessageHeader,
+                    Headers.OriginatingSagaId,
+                    Headers.RelatedTo,
+                    Headers.ConversationId,
+                    Headers.MessageIntent,
+                    "NServiceBus.ExceptionInfo.ExceptionType",
+                    "NServiceBus.ExceptionInfo.HelpLink",
+                    "NServiceBus.ExceptionInfo.Message",
+                    "NServiceBus.ExceptionInfo.Source",
+                    "NServiceBus.ExceptionInfo.StackTrace"
+                },
+                SensitiveHeaders = new[]
+                {
+                    Headers.RouteTo,
+                    Headers.DestinationSites,
+                    Headers.OriginatingSite,
+                    Headers.HttpTo,
+                    Headers.ReplyToAddress,
+                    Headers.NServiceBusVersion,
+                    Headers.ReturnMessageErrorCodeHeader,
+                    Headers.IsSagaTimeoutMessage,
+                    Headers.IsDeferredMessage,
```

Comment:
With native delayed deliveries this is becoming less and less useful, still makes no sense to modify this header